neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -53,48 +54,103 @@ protected override void OnReceive(object message)
             switch (msg.Command)
             {
                 case "addr":
-                    OnAddrMessageReceived(msg.Payload.AsSerializable<AddrPayload>());
-                    break;
+                    {
+                        var payload = msg.Payload.AsSerializable<AddrPayload>();
+                        if (!Plugin.ReceivedMessageAllowed(msg.Command, payload)) return;
+                        OnAddrMessageReceived(payload);
+                        break;
+                    }
                 case "block":
-                    OnInventoryReceived(msg.Payload.AsSerializable<Block>());
-                    break;
+                    {
+                        var payload = msg.Payload.AsSerializable<Block>();
+                        if (!Plugin.ReceivedMessageAllowed(msg.Command, payload)) return;
+                        OnInventoryReceived(payload);
+                        break;
+                    }
                 case "consensus":
-                    OnInventoryReceived(msg.Payload.AsSerializable<ConsensusPayload>());
-                    break;
+                    {
+                        var payload = msg.Payload.AsSerializable<ConsensusPayload>();
+                        if (!Plugin.ReceivedMessageAllowed(msg.Command, payload)) return;
+                        OnInventoryReceived(payload);
+                        break;
```

Comment:
All consensus messages pass here Shargon... perhaps there's no need to change Consensus itself. What do you think?

Reply:
I want to take the info of the payload (deserialized) :P

Reply:
After deserialization we can lose information, yes?
It sounds good to have it deserialized.

But what Igor said is that we may not need this in the switch `(message.Type)`.
I think it makes sense because the ReceivedMessageAllowed can process it and perform the switch inside it.

Reply:
@igormcoelho, any other insight here? I think that we could proceed with this plugin as soon as possible.